backend/api/models.py

```diff
@@ -31,6 +31,7 @@ class Product(models.Model):
   price.help_text = 'In cents'
   image = models.ImageField(upload_to='uploaded_images/')
   visible = models.BooleanField(default=True)
+  last_bought = models.DateTimeField(null=True, default=None)
```

Comment:
This info is already available in the transactions table, so instead of adding a new column we could add a method to product that looks for the most recent transaction for this product and returns the date. 

Of course doing it with a new column works like a cache table so lookups are faster (not by a lot though). I would prefer not duplicating data this way.

Reply:
As noted in https://github.com/radiodiodi/sahkopiikki/pull/1/commits/cd0d47387538a068790b98a9a25c0a01f0928694 this would be a non problem if we were using PostgreSQL database. 

I could not figure out an elegant non-spaghetti solution to this. AFAIK model methods can't be used for queryset ordering. One solution would be to reorder in front-end but low-end client devices would suffer.

Reply:
I think we ran into this kind of problem on SIK web2.0, when we tried to sort members based on their last membership payment.

Ultimately we found a solution that uses member method to return a queryset with member model data, appended with last payment column.

Reply:
def get_members_with_latest_payment(members_query):
        """Return QuerySet of given members QS with last_paid attribute."""
        latest = Payment.objects.filter(member=OuterRef('pk')).order_by('-date')
        return members_query.annotate(last_paid=Subquery(latest.values('date')[:1]))

Reply:
That solution worked with both PostgreSQL and SQLite

Reply:
The problem is not the inclusion of latest purchase date to the results but the ordering of the results. The solution proposed in the PR would add a minuscule amount of duplicate data (without any risk of notable side effects) but would make queries bit faster as `transactions` need not to be joined. Both technical benefits and drawbacks are IMHO irrelevant in this case.

One solution would be to remove duplicates from `Products.objects.all().order_by('-transaction__timestamp')` with a list operation but that would get slower and slower with time as every new transaction creates a new duplicate.

Or then just implement the method suggested by @jantuomi, include it in to results and reorder in front.

Reply:
So either we move to another DB or go with this solution. I say let's go with this for now and think about a new DB later.